File: Java/xunit-test-patterns/src/test/java/_07_picking_test_values/_02_BoundaryValuesAndConstantTests.java
package _07_picking_test_values;

import _03_parameterized_tests.Person;
import lombok.val;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;
import org.junit.jupiter.api.Test;

import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

public class _02_BoundaryValuesAndConstantTests {
    public static Stream<Arguments> shouldBeAdultDependingOnAgeData() {
        return Stream.of(
          Arguments.of(Person.ADULT_AGE - 1, false),
          Arguments.of(Person.ADULT_AGE, true)
        );
    }

    @ParameterizedTest
    @MethodSource("shouldBeAdultDependingOnAgeData")
    public void shouldBeAdultDependingOnAge(int age, boolean expectedIsAdult) {
        //GIVEN
        val person = new Person(age);

        //WHEN
        val isAdult = person.isAdult();

        //THEN
        assertThat(isAdult).isEqualTo(expectedIsAdult);
    }

    @Test
    public void shouldSayThatAdultAgeIs18() {
        assertThat(Person.ADULT_AGE).isEqualTo(18);
    }

}

File: CSharp/XUnitTestPatterns/XUnitTestPatterns/_06_PickingTestValues/_02_BoundaryValuesAndConstantTests.cs
using NUnit.Framework;
using XUnitTestPatterns._03_ParameterizedTests;

namespace XUnitTestPatterns._06_PickingTestValues
{
  public class _02_BoundaryValuesAndConstantTests
  {
    [TestCase(Person.AdultAge, true)]
    [TestCase(Person.AdultAge - 1, false)]
    public void ShouldBeAdultDependingOnAge(int age, bool expectedIsAdult)
    {
      //GIVEN
      var person = new Person(age);

      //WHEN
      var isAdult = person.IsAdult();

      //THEN
      Assert.AreEqual(expectedIsAdult, isAdult);
    }

    [Test]
    public void ShouldDefineAdultAgeAs18()
    {
      Assert.AreEqual(18, Person.AdultAge);
    }

  }
}